pkg/ring/kv/metrics.go

```diff
@@ -32,12 +22,36 @@ func errorCode(err error) string {
 }
 
 type metrics struct {
-	c Client
+	c               Client
+	requestDuration *instrument.HistogramCollector
+}
+
+func newMetricsClient(name string, backend string, c Client, reg prometheus.Registerer) Client {
+	// TODO: Ensure all clients are passed a valid registry and remove the following check
+	if reg == nil {
+		reg = prometheus.DefaultRegisterer
+	}
+
+	return &metrics{
+		c: c,
+		requestDuration: instrument.NewHistogramCollector(
+			promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+				Namespace: "cortex",
+				Name:      "kv_request_duration_seconds",
+				Help:      "Time spent on kv store requests.",
+				Buckets:   prometheus.DefBuckets,
+				ConstLabels: prometheus.Labels{
+					"name": name,
+					"type": backend,
+				},
```

Comment:
Please mention this change in the CHANGELOG. You should also check if it has an impact on the cortex mixin's dashboards (depends on how we group series, I haven't checked).